pkg/controller.go

```diff
@@ -129,3 +133,24 @@ func (c *ExternalIpController) processServiceExternalIPs(service *v1.Service) {
 		c.queue.Add(&netutils.AddCIDR{cidr})
 	}
 }
+
+func (c *ExternalIpController) deleteServiceExternalIPs(service *v1.Service, store cache.Store) {
+	if len(service.Spec.ExternalIPs) == 0 {
+		return
+	}
+	ips := make(map[string]bool)
+	// collect external IPs of existing services
+	for s := range store.List() {
+		if store.List()[s].(*v1.Service).ObjectMeta.UID != service.ObjectMeta.UID {
```

Comment:
calling List every time to get single object is inefficient, in this case lets just dispatch this object right here...
Also please don't cast object to concrete type every time you need it. Do it once if you have to
```
_, svcInt := range store.List {
    svc := svcInt.(*v1.Service)
}
```
Lets avoid UID please, in store namespace/name is used as unique key, and u can get this key from any type by using KeyFunc of that store (i think we discussed that)